File: Iteration 2/Java Code/test/unit/InstructorTest.java
package test.unit;

import static org.junit.Assert.assertEquals;

import java.util.Calendar;
import java.util.GregorianCalendar;

import org.junit.Test;

import Enums.InstructorType;
import IDs.InstructorID;
import lecture.Schedule;
import person.Instructor;

public class InstructorTest {

	@Test
	public void test_instructor_InstructorID() {
	
		InstructorID testInstructorID = new InstructorID(1,2);
		Calendar testDateOfEntry = new GregorianCalendar(2011,10,28);
		InstructorType testInstructorType = InstructorType.Assistant;
		Schedule testSchedule = new Schedule(null, null, null);
		
		Instructor testInstructor = new Instructor("a","b",testInstructorID,testDateOfEntry,testInstructorType,testSchedule);
		
		assertEquals(testInstructorID, testInstructor.getID());

	}
	@Test
	public void test_instructor_scheudle() {
	
		InstructorID testInstructorID = new InstructorID(1,2);
		Schedule testSchedule = new Schedule(null, null, null);
		Calendar testDateOfEntry = new GregorianCalendar(2011,10,28);
		InstructorType testInstructorType = InstructorType.Assistant;
		
		Instructor testInstructor = new Instructor("a","b",testInstructorID,testDateOfEntry,testInstructorType,testSchedule);
		
		testSchedule.setPerson(testInstructor);
		
		assertEquals(testSchedule, testInstructor.getSchedule());

	}
	@Test
	public void test_instructor_date_of_entry() {
	
		InstructorID testInstructorID = new InstructorID(1,2);
		Schedule testSchedule = new Schedule(null, null, null);
		Calendar testDateOfEntry = new GregorianCalendar(2011,10,28);
		InstructorType testInstructorType = InstructorType.Assistant;
		
		Instructor testInstructor = new Instructor("a","b",testInstructorID,testDateOfEntry,testInstructorType,testSchedule);
		
		assertEquals(null, testInstructor.getDateOfEntry());

	}
	@Test
	public void test_instructor_instructor_type() {
	
		InstructorID testInstructorID = new InstructorID(1,2);
		Schedule testSchedule = new Schedule(null, null, null);
		Calendar testDateOfEntry = new GregorianCalendar(2011,10,28);
		InstructorType testInstructorType = InstructorType.Assistant;
		
		Instructor testInstructor = new Instructor("a","b",testInstructorID,testDateOfEntry,testInstructorType,testSchedule);
	
		assertEquals(InstructorType.Assistant, testInstructor.getInstructorType());
		
	}
}

File: CSE3063F22P1_GRP1_iteration3/pythonCode/InstructorTest.py
#!/usr/bin/env python
from InstructorID import InstructorID
from InstructorType import InstructorType
from Schedule import Schedule
from Instructor import Instructor
import datetime
import unittest
class InstructorTest(unittest.TestCase):
    # generated source for class InstructorTest 
    def test_instructor_InstructorID(self):
        # generated source for method test_instructor_InstructorID 
        testInstructorID = InstructorID(1, 2)
        testInstructor = Instructor()
        testInstructor.setID(testInstructorID)
        self.assertEqual(testInstructorID, testInstructor.getID())
    def test_instructor_scheudle(self):
        # generated source for method test_instructor_scheudle 
        testSchedule = Schedule()
        testInstructor = Instructor()
        testSchedule.setPerson(testInstructor)
        testInstructor.setSchedule(testSchedule)
        self.assertEqual(testSchedule, testInstructor.getSchedule())
    def test_instructor_date_of_entry(self):
        # generated source for method test_instructor_date_of_entry 
        testDateOfEntry = datetime.date(2011, 10, 28)
        testInstructor = Instructor()
        testInstructor.setDateOfEntry(testDateOfEntry)
        self.assertEqual(testDateOfEntry, testInstructor.getDateOfEntry())
    def test_instructor_instructor_type(self):
        # generated source for method test_instructor_instructor_type 
        testInstructorType = InstructorType.Assistant
        testInstructor = Instructor()
        testInstructor.setInstructorType(testInstructorType)
        self.assertEqual(InstructorType.Assistant, testInstructor.getInstructorType())
if __name__ == '__main__':
    unittest.main()
    